staging/src/k8s.io/client-go/util/workqueue/delaying_queue_test.go

```diff
@@ -69,6 +285,247 @@ func TestSimpleQueue(t *testing.T) {
 	}
 }
 
+// TestImmediateAdd calls to Add() should 'promote' items out of waiting (i.e. delete from waiting)
+// and immediately add them (via the waitingLoop)
+func TestImmediateAdd(t *testing.T) {
+	fakeClock := testingclock.NewFakeClock(time.Now())
+	q := NewTestDelayingQueueWithCustomClock(t, fakeClock, "")
+
+	first := "foo"
+	second := "foo"
```

Comment:
I know the reason behind this is probably to distinguish items in active vs waiting queue, but I still find the naming confusing. I would expect the first and second to be different keys. 

I think the asserts down bellow would still give the reader clearer signal what key(object) is in which queue if they had only 1 name.

The duplication case should be obvious as well - the item is there twice - once in active, once in waiting queue.

Reply:
I don't want to change the style that is used on all of the tests because this will be confusing to other developers and I really don't want to rewrite the existing tests as these are showing that the original behaviour still works -and if I update all the tests then I could be masking issues with the backwards compatibility.

Reply:
yeah, let's not touch the existing tests, but in there, we used first/second/third to indicate a new item. In the new tests it is in the end the same item, because it is the same key. I think it would help readability if we used the variables as keys as well. We can explain in comments if something is not clear, but I am not seeing anything especially confusing at this time. 

Explanation example: `// we expect the same item to be present in the active and waiting queue because we used PermitActiveAndWaiting`